### PR TITLE
GrafanaUI: Add aria-label prop to RadioButtonGroup

### LINKED ISSUE #66475: Radio Button Group does not have an associated label prop which is required by screen readers
### User Experience

Screen reader users will get impacted if the label of any control or radio buttons are not announced by the screen reader. Users will find it difficult to understand the control purpose if label is not announced and not able to take actions on controls.

### Repro steps

Go to the [RadioButtonGroup](https://developers.grafana.com/ui/latest/index.html?path=/story/forms-radiobuttongroup--radio-buttons) in the Storybook UI. Turn on your screen reader. Observe that no label is read out associated with the radio options when you first enter them.

e.g. In this case the screen reader would need to read out "Prometheus, selected, radio button, 1 of 3, **_Full width, group_**" _(currently it doesn't read out Full width)_
image

A practical example in the application:
https://user-images.githubusercontent.com/6906380/231788398-d4016577-8619-45bb-ba64-933ee0fb27ac.mov

### Severity
`SEV 2`

### MS Sheet ID
[256757](https://docs.google.com/spreadsheets/d/1kB-Sz3lvfje93HLVIx4UhDvwgU-J7k1PVN70ryQkc4M/edit#gid=888389206&range=A4)

### PR DESCRIPTION
**Pull request by khushijain21**
**What is this feature?**

This feature adds role and area-label on the div wrapping the Radio Button for better accessbility

**Which issue(s) does this PR fix?**:

This PR fixes  #66475

### CODE REVIEW

**Comment by tskarhed** [comment]
We are currently thinking a little bit about the approach here. It should be accessible with the `Field` component, but is rarely used that way. Another option is to make it into a fieldset. And a third option is to add a aria-label, as in this PR.

**Comment by tonypowa** [comment]
hi @tskarhed 
Are there any news regarding this pull request that you can share with the contributor?
thank you!

**Review by tskarhed — APPROVED** [review]

**Comment by tskarhed** [comment]
@khushijain21 Hi, could you sync this with the main branch?

File: packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.tsx [hunk 1]
```diff
@@ -36,6 +37,7 @@ export function RadioButtonGroup<T>({
   className,
   fullWidth = false,
   autoFocus = false,
+  ariaLabel
```
Comment (tskarhed) [inline packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.tsx:40]:
```suggestion
  ariaLabel,
```
Forgot to address this. A good reminder to use Prettier!

Comment (tskarhed) [reply packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.tsx:40]: @khushijain21 Let's make the CI happy :)

Comment (khushijain21) [reply packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.tsx]: still shows error, can you help?

Comment (tskarhed) [reply packages/grafana-ui/src/components/Forms/RadioButtonGroup/RadioButtonGroup.tsx]: Ah, yes, they have updated so now it should be `aria-label` instead.

**Review by tskarhed — CHANGES_REQUESTED** [review]
The CI is still complaining about aria label. Could you run this locally to ensure that the naming satisfies Typescript and Javascript conventions?

**Comment by joshhunt** [comment]
I've renamed the aria-label prop to match our preferred convention. Once this passes CI, we can merge.

**Review by tskarhed — APPROVED** [review]

**Pull request merged**